### PR TITLE
vault: remove native staking mock

### PR DESCRIPTION
**Pull request by uint**
~~WIP~~ (done). Closes part of #95.

### CODE REVIEW

File: contracts/provider/vault/src/multitest.rs [hunk 1]
```diff
@@ -494,12 +535,8 @@ fn stake_local() {
...
     );
 
-    local_staking
-        .unstake(
-            vault.contract_addr.to_string(),
-            user.to_owned(),
-            coin(100, OSMO),
-        )
+    proxy
+        .unstake(user.to_owned(), coin(100, OSMO))
```
Comment (uint) [inline contracts/provider/vault/src/multitest.rs:539]: @maurolacy I can't manage to simulate an unstake with the real native staking contract. It looks like the real thing has an unbonding period and something external (a native Cosmos SDK module maybe?) sends it money. Is that what happens? Do we have a way to do that in tests?

Comment (maurolacy) [reply contracts/provider/vault/src/multitest.rs:539]: I think you may be hitting a current limitation in multi-test. Take a look at the `ProcessQueue` "sudo" message. There are some examples in the `native-staking-proxy` contract tests.

Comment (maurolacy) [reply contracts/provider/vault/src/multitest.rs:539]: There's also a cw-mt issue to improve on this by the way. See https://github.com/CosmWasm/cw-multi-test/issues/53.

Comment (uint) [reply contracts/provider/vault/src/multitest.rs:539]: Yeah, I think I figured it out. I stumbled into `ProcessQueue` too.

File: contracts/provider/vault/src/multitest.rs [hunk 2]
```diff
@@ -28,17 +28,52 @@ const SLASHING_PERCENTAGE: u64 = 10;
...
+    let mut app = MtApp::new(|router, _api, storage| {
         for (&user, amount) in std::iter::zip(users, amounts) {
             router
                 .bank
                 .init_balance(storage, &Addr::unchecked(user), coins(*amount, OSMO))
                 .unwrap();
         }
     });
+
+    set_chain_native_denom(&mut app, OSMO);
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:40]: 👍🏼 I would just put it in place here, but not a bad idea to encapsulate it.

**Review by maurolacy — APPROVED** [review]
LGTM. Thanks for refactoring it.

**Review by JakeHartnell — APPROVED** [review]
:fire:

**Pull request merged**